camply/containers/data_containers.py

```diff
@@ -20,6 +22,30 @@ class SearchWindow(CamplyModel):
     Search Window for Campsite Search
     """
 
+    @validator("start_date")
+    def start_date_must_be_in_future(self, v):
+        """
+        Validate that start_date is in the future
+        """
+        current_date = datetime.datetime.now().date()
+        print(v, current_date)
+        if v < current_date:
+            raise ValueError("must be in the future")
+
+        return v
```

Comment:
[Validators use `cls` instead of `self`](https://pydantic-docs.helpmanual.io/usage/validators/) - kind of confusing but that's what's throwing the error here: https://github.com/juftin/camply/actions/runs/3576943050/jobs/6015349570#step:8:427

I like the idea of validating the start date at the `SearchWindow` level - currently I coerce the `SearchWindow` into being compliant with the current date here: https://github.com/juftin/camply/blob/173ee4792dbee1fb922438477697780631986932/camply/search/base_search.py#L542-L569

If we need to set this data validation at the `SearchWindow` level these validators should coerce the dates into >= current date instead of raising an Error. 

I kind of like the idea of being able to provide a start date in the past as long as the end date is in the future without raising an error. What do you think?

Reply:
Yeah that makes sense; I'd prefer it to be more tolerant of bad input.